### ISSUE TITLE
Clarify requirements for Updates query

### ISSUE THREAD

**Issue by sebastienros** [TechEmpower/FrameworkBenchmarks#4773]
Before opening issues related to several Updates implementations, I'd like to get confirmation on the requirements.

Reading them here: https://github.com/TechEmpower/FrameworkBenchmarks/wiki/Project-Information-Framework-Tests-Overview#database-updates

> viii. Each World object must be persisted to the database with its new randomNumber value.
ix. Use of batch updates is acceptable but not required. To be clear: batches are not permissible for selecting/reading the rows, but batches are acceptable for writing the updates.

Is it expected that when using a batch query, it should contain `N` update statements, 
or is it accepted to dynamically construct a single update statement using either `case/when` statement or temporary tables containing all distinct `id` and `randomNumber` values?

**Comment by nbrady-techempower** [comment]
That's a good question. I don't think we have anyone using `case/when` updates in a single statement but now I do remember seeing a concatenated update statement somewhere.

It's hard to put all the things you _can't_ do in these requirements. I'd say you can only do what's explicitly allowed; in this case **multiple update statements**, **batches**, **transactions**.

Let's clarify, though. @bhauer @msmith-techempower @michaelhixson Thoughts here? Concatenated update statements? `case/when` single update statements for multiple updates?

I'll update the docs accordingly.

**Comment by michaelhixson** [comment]
My understanding is that a single statement that updates multiple rows is fine.

**Comment by sebastienros** [comment]
FYI Actix is using a single UPDATE statement like this:

```
UPDATE world SET randomnumber = temp.randomnumber 
FROM (VALUES (1, 1232), (2, 4563)) ORDER BY 1) AS temp(id, randomnumber) 
WHERE temp.id = world.id
```
with the random values directly in the sql statement.

Jooby2 is doing the same query, but with parameters `($1, $2)`. Still a single update statement.

Drogon-raw is using a when/case in a single update statement like this:

```
update world set randomnumber=case id
when $1 then $2
when $3 then $4
else randomnumber end where id in ($5, $6)
```

So as long as you accept a single update statement these are fine. Maybe this could be clarified in the specs, as this will be the best way to update with raw db access.

**Comment by fafhrd91** [comment]
i use same approach in actix as in [undertow](https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/frameworks/Java/undertow/src/main/java/hello/HelloWebServer.java#L179)

**Comment by fafhrd91** [comment]
rust's pg driver does not support parameters for this query.

**Comment by michaelhixson** [comment]
At this point, I forget which was the first framework that we noticed using the "UPDATE...FROM" syntax.  It might have been h2o.  We noticed someone with really good updates results, wondered if they were cheating, turned on query logging to see what they were doing, and decided they were not cheating.  I changed Undertow's code to use that approach after that discussion.

Would this rephrasing of that requirement make things clearer?

> Using bulk updates&mdash;batches of update statements or individual update statements that affect multiple rows&mdash;is acceptable but not required.  To be clear: bulk reads are not permissible for selecting/reading the rows, but bulk updates are acceptable for writing the updates.
